phoenix-core/src/main/java/org/apache/phoenix/coprocessor/IndexRebuildRegionScanner.java

```diff
@@ -465,17 +490,32 @@ private void verifyIndexRows(ArrayList<KeyRange> keys) throws IOException {
                 if (dataPut == null) {
                     exceptionMessage = "Index verify failed - Missing data row - " + indexHTable.getName();
                     String errorMsg = "Missing data row";
-                    logToIndexToolOutputTable(null, result.getRow(), 0, getMaxTimestamp(result), errorMsg);
+                    logToIndexToolOutputTable(null, result.getRow(), 0, getMaxTimestamp(Arrays.asList(result.rawCells())), errorMsg);
                     if (!onlyVerify) {
                         throw new IOException(exceptionMessage);
                     }
                 }
                 verifySingleIndexRow(result, dataPut);
+                indexRowKeys.remove(result.getRow());
                 rowCount++;
             }
         } catch (Throwable t) {
             ServerUtil.throwIOException(indexHTable.getName().toString(), t);
         }
+        // Check if any expected rows from index(which we didn't get) are already expired due to TTL
+        // TODO: metrics for expired rows
```

Comment:
Could you please also create a jira for this so that we don't lose it in the comments? Thanks.

Reply:
Sure, will do.

Reply:
It would be nice to link it to the code's comment.